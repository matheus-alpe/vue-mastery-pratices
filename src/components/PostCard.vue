<script setup>
defineProps({
    title: {
        type: String,
        required: true,
    },
    body: {
        type: String,
        required: true,
    },
});
</script>

<template>
    <div>
        <h1>{{ title }}</h1>
        <p>{{ body }}</p>
    </div>
</template>

<style>
h1 {
    font-size: 32px;
    font-weight: 700;
    color: #fff;
}
</style>
