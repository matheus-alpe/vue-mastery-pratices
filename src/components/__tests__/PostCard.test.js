import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import PostCard from '../PostCard.vue'

describe('PostCard', () => {
  test('created posts render correctly', async () => {
    const wrapper = mount(PostCard, {
      props: {
        title: 'Post title',
        body: 'Post body',
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
