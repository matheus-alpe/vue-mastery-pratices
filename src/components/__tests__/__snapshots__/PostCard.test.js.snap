// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostCard > created posts render correctly 1`] = `
"<div tile=\\"Post title\\">
  <h1></h1>
  <p>Post body</p>
</div>"
`;
